zstacklib/zstacklib/utils/file_system_upload_task.py

```
@@ -0,0 +1,111 @@
+
+    def create_file_if_not_exists(self):
+        if self.task_created:
+            return
+
+        with lock.NamedLock("upload-file-task-%s" % self.taskUuid):
+            if not self.task_created:
+                try:
+                    size = int(self.expectedSize)
+                except (TypeError, ValueError):
+                    raise Exception("expectedSize must be a valid integer, current: %s (type: %s)"
+                                    % (self.expectedSize, type(self.expectedSize).__name__))
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)